SwiftWisdom/Core/StandardLibrary/String/String+Indexing.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  String+Substrings.swift
+//  ChristmasCheer
+//
+//  Created by Logan Wright on 10/27/15.
+//  Copyright © 2015 lowriDevs. All rights reserved.
+//
+
+import Foundation
+
+public extension String {
+    // TODO: Need to write tests for this
+
+    public subscript(range: Range<Int>) -> String {
+        let chars = Array(characters)
+        let substringCharacters = chars[range]
+        return String(substringCharacters)
+    }
+
+    public subscript(range: CountableClosedRange<Int>) -> String {
+        let chars = Array(characters)
+        let substringCharacters = chars[range]
+        return String(substringCharacters)
+    }
+
+    public mutating func dropFirst() {
+        guard !isEmpty else { return }
+        self = self[1..<characters.count]
+    }
+
+    // From http://stackoverflow.com/questions/25138339/nsrange-to-rangestring-index/32379600#32379600
+    func ip_range(fromNSRange nsRange: NSRange) -> Range<String.Index>? {
```

Comment:
i dont think we need fromNSRange here... fromRange would do. type is inferred by compiler.